### ISSUE TITLE
visual problem

### ISSUE THREAD

**Issue by darewolf007** [darewolf007/Python_lib#2]
## get ycbmodel from .xyz file
def get_model(item):
    input_file = open('/home/haowen/my_code/DenseFusiontorch17-master/datasets/ycb/YCB_Video_Dataset/models/{0}/points.xyz'.format(item))
    model = []
    while 1:
        input_line = input_file.readline()
        if not input_line:
            break
        input_line = input_line[:-1].split(' ')
        model.append([float(input_line[0]), float(input_line[1]), float(input_line[2])])
    model = np.array(model)
    input_file.close()
    return model
def transform(pred_r,pred_t,model_points):
    pred_r=torch.tensor(pred_r)
    pred_t=torch.tensor(pred_t)
    model_points=torch.tensor(model_points)
    pred_r = pred_r.view(1, 1, -1)
    pred_t = pred_t.view(1, 1, -1)
    bs, num_p, _ = pred_r.size()
    pred_r = pred_r / (torch.norm(pred_r, dim=2).view(bs, num_p, 1))
    base = torch.cat(((1.0 - 2.0 * (pred_r[:, :, 2] ** 2 + pred_r[:, :, 3] ** 2)).view(bs, num_p, 1), \
                      (2.0 * pred_r[:, :, 1] * pred_r[:, :, 2] - 2.0 * pred_r[:, :, 0] * pred_r[:, :, 3]).view(bs,
                                                                                                               num_p,
                                                                                                               1), \
                      (2.0 * pred_r[:, :, 0] * pred_r[:, :, 2] + 2.0 * pred_r[:, :, 1] * pred_r[:, :, 3]).view(bs,
                                                                                                               num_p,
                                                                                                               1), \
                      (2.0 * pred_r[:, :, 1] * pred_r[:, :, 2] + 2.0 * pred_r[:, :, 3] * pred_r[:, :, 0]).view(bs,
                                                                                                               num_p,
                                                                                                               1), \
                      (1.0 - 2.0 * (pred_r[:, :, 1] ** 2 + pred_r[:, :, 3] ** 2)).view(bs, num_p, 1), \
                      (-2.0 * pred_r[:, :, 0] * pred_r[:, :, 1] + 2.0 * pred_r[:, :, 2] * pred_r[:, :, 3]).view(bs,
                                                                                                                num_p,
                                                                                                                1), \
                      (-2.0 * pred_r[:, :, 0] * pred_r[:, :, 2] + 2.0 * pred_r[:, :, 1] * pred_r[:, :, 3]).view(bs,
                                                                                                                num_p,
                                                                                                                1), \
                      (2.0 * pred_r[:, :, 0] * pred_r[:, :, 1] + 2.0 * pred_r[:, :, 2] * pred_r[:, :, 3]).view(bs,
                                                                                                               num_p,
                                                                                                               1), \
                      (1.0 - 2.0 * (pred_r[:, :, 1] ** 2 + pred_r[:, :, 2] ** 2)).view(bs, num_p, 1)),
                     dim=2).contiguous().view(bs * num_p, 3, 3)

    base = base.contiguous().transpose(2, 1).contiguous()
    model_points = model_points.view(1, -1, 3)

    pred_t = pred_t.contiguous().view(1, 1, 3)
    pred = torch.add(torch.bmm(model_points, base), pred_t)[0]
    return pred
def project_points(cld,pred_rt):
    cam_cx = 426.0710754394531
    cam_cy = 240.34532165527344
    cam_fx = 426.2512512207031
    cam_fy = 426.2512512207031
    # cam_cx = 312.9869
    # cam_cy = 241.3109
    # cam_fx = 1066.778
    # cam_fy = 1067.487
    pred_t=np.array([pred_rt[0][3], pred_rt[1][3], pred_rt[2][3]])
    # pred_model = transform(pred_rt[:4],pred_rt[4:],cld)
    pred_model=np.add(np.dot(cld,pred_rt[:3, :3].T),pred_t)
    pred_model_x=pred_model[:,0]
    pred_model_y=pred_model[:,1]
    pred_model_z=pred_model[:,2]
    x = ((pred_model_x * cam_fx) / pred_model_z) + cam_cx
    y = ((pred_model_y * cam_fy) / pred_model_z) + cam_cy
    project_xy=np.concatenate((x[:,None],y[:,None]),axis=1)
    return x,y
def visual_pic(project_x,project_y,pic):
    # from PIL import Image
    # project_pic = Image.fromarray(np.uint8(project_xy))
    # rgb_pic=Image.fromarray(np.uint8(pic[0]))
    # Image.blend(rgb_pic, project_pic, alpha=0.5)
    import matplotlib.pyplot as plt
    plt.scatter(project_x,project_y, alpha=0.6)
    plt.imshow(pic, alpha=0.4)
    plt.show()
